pandas/tests/frame/methods/test_sort_values.py

```diff
@@ -868,3 +868,25 @@ def test_sort_values_pos_args_deprecation(self):
             result = df.sort_values("a", 0)
         expected = DataFrame({"a": [1, 2, 3]})
         tm.assert_frame_equal(result, expected)
+
+    def test_sort_values_validate_ascending(self):
+        # GH41634
+        df = DataFrame({"D": [23, 7, 21]})
+        indexer = df["D"].argsort().values
+        r_indexer = indexer[::-1]
+
+        msg = 'For argument "ascending" expected type bool, received type str.'
+        with pytest.raises(ValueError, match=msg):
+            df.sort_values(by="D", ascending="False")
+
+        expected_df = df.loc[df.index[r_indexer]]
```

Comment:
Please call them expected and result here too

lgtm otherwise

Reply:
Thanks for all the comments, Patrick. This is resolved.